### ISSUE TITLE
Type hint errors in new `azure-storage-blob` release

### ISSUE THREAD

**Issue by anders-kiaer** [Azure/azure-sdk-for-python#20771]
- **Package Name**: azure-storage-blob
- **Package Version**: 12.9.0  
- **Operating System**: Ubuntu
- **Python Version**: Python 3.8

**Describe the bug**
Type hint issues. We get the following output from `mypy` after the `azure-storage-blob` version released yesterday:
```
[...]/azure/storage/blob/_serialize.py:129: error: Type signature has too many arguments  [syntax]
[...]/azure/storage/blob/_blob_client.py:1407: error: Type signature has too few arguments  [syntax]
[...]/azure/storage/blob/aio/_blob_client_async.py:808: error: Type signature has too few arguments  [syntax]
error: "azure.storage.blob.aio.ContainerClient" has no attribute "__aenter__"  [attr-defined]
error: "azure.storage.blob.aio.ContainerClient" has no attribute "__aexit__"  [attr-defined]
```

Code references to the first three:
https://github.com/Azure/azure-sdk-for-python/blob/da36d3091177ca3918c93977296d33372b34aa5c/sdk/storage/azure-storage-blob/azure/storage/blob/_serialize.py#L129-L130
https://github.com/Azure/azure-sdk-for-python/blob/da36d3091177ca3918c93977296d33372b34aa5c/sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py#L1407-L1408
https://github.com/Azure/azure-sdk-for-python/blob/da36d3091177ca3918c93977296d33372b34aa5c/sdk/storage/azure-storage-blob/azure/storage/blob/aio/_blob_client_async.py#L808-L809

**To Reproduce**
Use `mypy` on newly released `azure-storage-blob==12.9.0`

For the two last type hint issues listed above you will need to use `azure.storage.blob.aio.ContainerClient` within `async with` in the downstream/user script to trigger the warning, e.g.:
```python
from azure.storage.blob.aio import ContainerClient

async with ContainerClient.from_connection_string(some_string) as container_client:
    ...
```

**Expected behavior**
No type issues

**Comment by xiafu-msft** [comment]
Hi @anders-kiaer 

Thanks for reporting this! We are putting out a pr to fix this.

**Comment by jalauzon-msft** [comment]
Hi @anders-kiaer (and others), apologies for the delay in addressing this. Two of the errors were recently fixed in #22741 and I just created #23098 to address another. These will go in our next release (expected in March).

The `__aenter__` and `__aexit__` ones are more interesting since `ContainerClient` does have these methods through its inheritance of `AsyncStorageAccountHostsMixin`. I'm not terribly familiar with `mypy` so I wonder if it is supposed to pick these up through inheritance?

**Comment by jalauzon-msft** [comment]
I looked into the `__aenter__` and `__aexit__` issues and these seem to be caused by an issue with `mypy` (or one of its dependencies) that was addressed in v0.812 (see python/mypy#9739). I have verified these issues are not present with the provided code sample using `mypy==0.931`.
